### DISCUSSION TITLE
RSS feed doesn't produce valid links

### DISCUSSION THREAD

**Discussion by jmichelgarcia** [google/docsy#1518] [Q&A] [answered]
Hi folks 👋,

Looking at the [example site RSS](https://example.docsy.dev/blog/index.xml), I see the produced URL (under the `link` attribute) is not a compliant url (it's missing the base url).

Is this expected? I will keep investigating on my end in the meantime.

```xml
<title>Goldydocs – Docsy Blog</title>
<link>/blog/</link>
<description>Recent content in Docsy Blog on Goldydocs</description>
<generator>Hugo -- gohugo.io</generator>
<atom:link href="/blog/index.xml" rel="self" type="application/rss+xml"/>
<item>
<title>Blog: Easy documentation with Docsy</title>
<link>/blog/2018/10/06/easy-documentation-with-docsy/</link>
<pubDate>Sat, 06 Oct 2018 00:00:00 +0000</pubDate>
<guid>/blog/2018/10/06/easy-documentation-with-docsy/</guid>
<description>
```

<img width="872" alt="Screenshot 2023-05-18 at 14 44 24" src="https://github.com/google/docsy/assets/2745523/b9f3d7d1-767c-450d-b825-11f9d54ea69d">

**Comment by fekete-robert** [comment]
It looks the same on the main site as well: https://www.docsy.dev/blog/index.xml

**Comment by jmichelgarcia** [comment]
Am I missing something and those links are semantically correct?
Or are you saying this is `broken` everywhere?

**Comment by fekete-robert** [comment]
I'm not using RSS anywhere, so I'm not sure about it, but looking at the xml of other Docsy-based sites (like https://github.com/istio/istio.io/blob/master/layouts/blog/rss.xml) I'd say it's broken everywhere. It seems that the istio site uses a custom layout file that explicitly adds the baseurl: https://github.com/istio/istio.io/blob/master/layouts/blog/rss.xml

**Comment by jmichelgarcia** [comment]
What's funny is that on `hugo serve` I get: `<link>//localhost:1313/en/docs...</link>`
But when serving from Render I get: `<link>/en/docs/....</link>`

**Comment by jmichelgarcia** [comment]
Might be my baseUrl setting, having a look at the docs and trying a few things out.

**Comment by aimeeu** [comment]
That's been broken for over a year... or at least that's when it was brought to my attention. I played around with the `baseURL` setting but never did get the RSS links to work.  If you find a solution, please share!

**Comment by jmichelgarcia** [comment, marked as answer]
My baseURL was set to "/", changing it to the correct domain solved my problem. Something like
`baseURL="https://domain.com/"`

Trailing slash is important.

**Reply by aimeeu** [reply]
Hmmm... I tried that approach a while ago but found that messed up the generated hyperlinks (`ref` shortcode) for local and also the deploy preview in Netlify.  IIRC, all the generated links used the baseURL (hard-coded domain) instead of using the env URL (`localhost` or the url generated by Netlify preview).   Is that still the case?

**Reply by jmichelgarcia** [reply]
I'm not seeing problems.

My local config uses `baseURL = "/"`
My PR config uses (I use Render) `env HUGO_BASEURL="${RENDER_EXTERNAL_URL}/" hugo --gc --minify`
My Prod config uses `baseURL = "https://domain.com/"`
